apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/Utilities.cs

```diff
@@ -43,6 +47,58 @@ internal static ByteString Concat(this ByteString left, ByteString right)
             return ByteString.CopyFrom(buffer);
         }
 
+        internal static ByteString RegexEscape(ByteString value)
+        {
+            // Logic taken from
+            // https://github.com/google/re2/blob/70f66454c255080a54a8da806c52d1f618707f8a/re2/re2.cc#L456
+            var result = new List<byte>(value.Length * 2);
+            foreach (byte b in value)
+            {
+                // Special handling for null chars.
+                if (b == 0)
+                {
+                    // Note that this special handling is not strictly required for RE2,
+                    // but this quoting is required for other regexp libraries such as
+                    // PCRE.
+                    // Can't use @"\0" since the next character might be a digit.
+                    result.AddRange(EscapedNullBytes);
+                }
+                else
+                {
+                    if (ShouldEscape(b))
```

Comment:
This will escape each byte within a UTF-8 sequence - which means we end up with invalid UTF-8. I doubt that that's the intention, is it? Shouldn't we be iterating in terms of characters rather than bytes?

Reply:
The `(b & 0x80) == 0x80` check in `ShouldEscape` should cover that. In any multi-byte pattern in UTF-8, all bytes have the high order bit set, so we'll never escape them.

Reply:
Ah, I'm with you. Yup, makes sense.